### PR TITLE
Constrain QCSubmit against QCElemental Pydantic v2 migration

### PR DESCRIPTION
**Pull request by mattwthompson**
This is yet another case of constraining 0.x upstreams against changes propagating out from the Pydantic v2 breaks

 Checklist

* [x] Used a static YAML file for the patch if possible ([instructions](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/main/recipe/README.md)).
* [x] Only wrote code directly into `generate_patch_json.py` if absolutely necessary.
* [x] Ran `pre-commit run -a` and ensured all files pass the linting checks.
* [x] Ran `python show_diff.py` and posted the output as part of the PR.
* [x] Modifications won't affect packages built in the future. 

```shell
$ python recipe/show_diff.py --subdirs noarch --use-cache              9:42:12  ☁  qcsubmit-qcelemental-pydantic-migration ☀ 𝝙
================================================================================
================================================================================
noarch
noarch::openff-qcsubmit-0.50.1-pyhd8ed1ab_0.conda
noarch::openff-qcsubmit-0.50.2-pyhd8ed1ab_0.conda
noarch::openff-qcsubmit-0.50.3-pyhd8ed1ab_0.conda
noarch::openff-qcsubmit-0.51.0-pyhd8ed1ab_0.conda
noarch::openff-qcsubmit-0.52.0-pyhd8ed1ab_0.conda
noarch::openff-qcsubmit-0.53.0-pyhd8ed1ab_0.conda
-    "qcelemental",
+    "qcelemental <=0.28",
```

### CODE REVIEW

**Comment by xhochy** [comment]
The diff in CI is not showing any changes.

**Comment by mattwthompson** [comment]
Hmm thanks ... will have a closer look

**Comment by mattwthompson** [comment]
The script to locally diff things isn't working as expected; I've tried a few dozen combinations of inputs and everything, including things like below with nearly no restrictions, and everything shows an empty diff locally. [This bit](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/main/recipe/README.md?plain=1#L211-L217) in the README leads me to believe it's not really meant to be trusted, so I'll be firing off some stuff to CI to see what works

```yaml
if:
  name: openff-qcsubmit
  timestamp_lt: 1728401565
  version_le: 0.53
then:
  - add_depends: espaloma

**Comment by mattwthompson** [comment]
Not sure what I'm doing wrong here
* I'm basing my changes off of something that worked previously (and not so long ago) https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/806/files
* Where in the CI logs would the diff be?

File: recipe/patch_yaml/openff-qcsubmit-qcelemental.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# This is another Pydantic v2 patch, one of our upstreams
+# (QCElemental) is undergoing their v1 -> v2 migration which will
+# likely result in API breaks
+if:
+  subdir_in: noarch
+  name: openff-qcsubmit-base
+  timestamp_lt: 1728073334
```
Comment (scaramallion) [inline recipe/patch_yaml/openff-qcsubmit-qcelemental.yaml]: The timestamp looks too short? The example in the README uses milliseconds.

Comment (mattwthompson) [reply recipe/patch_yaml/openff-qcsubmit-qcelemental.yaml]: That was it - thanks @scaramallion!

**Review by xhochy — APPROVED** [review]

**Comment by mattwthompson** [comment]
Thanks!

**Pull request merged**